### PR TITLE
G-API: Fix for "using cv makes G_TYPED_KERNEL fail"

### PR DESCRIPTION
**Pull request by akhakim**
…ilation

```
force_builders_only=Docs,linux

build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f
build_gapi_standalone:Custom Win=ade-0.1.1f
```

### CODE REVIEW

**Comment by anna-plyakina** [comment]
@dmatveev @TolyaTalamanov  @anton-potapov

File: modules/gapi/include/opencv2/gapi/gkernel.hpp [hunk 1]
```diff
@@ -233,7 +233,7 @@ class GKernelType<K, std::function<R(Args...)> >
 #define G_ID_HELPER_CLASS(Class)  Class##IdHelper
 
 #define G_ID_HELPER_BODY(Class, Id)                                         \
-    namespace detail                                                        \
+    namespace service                                                         \
```
Comment (TolyaTalamanov) [inline modules/gapi/include/opencv2/gapi/gkernel.hpp]: Align is broken

**Comment by TolyaTalamanov** [comment]
LGTM :+1:

File: doc/Doxyfile.in [hunk 2]
```diff
@@ -107,7 +107,7 @@ RECURSIVE              = YES
 EXCLUDE                = @CMAKE_DOXYGEN_EXCLUDE_LIST@
 EXCLUDE_SYMLINKS       = NO
 EXCLUDE_PATTERNS       = *.inl.hpp *.impl.hpp *_detail.hpp */cudev/**/detail/*.hpp *.m */opencl/runtime/* */legacy/* *_c.h @DOXYGEN_EXCLUDE_PATTERNS@
-EXCLUDE_SYMBOLS        = cv::DataType<*> cv::traits::* int void CV__* T __CV*
+EXCLUDE_SYMBOLS        = cv::DataType<*> cv::traits::* int void CV__* T __CV* service cv::service
```
Comment (alalek) [inline doc/Doxyfile.in]: Lets avoid that.

Please try to use this pattern:

```
#ifdef CV_DOXYGEN
... well documented macros without implementation or with documentation friendly stuff ...
#else
... real macro implementations (excluded from documentation) ...
#endif // CV_DOXYGEN
```

or approach from core/utils/trace.hpp (basically the same)

Comment (anna-plyakina) [reply doc/Doxyfile.in]: Applied.

**Comment by alalek** [comment]
> Merge pull request ...

Avoid using Pull requests in forks to sync with upstream repository.
[Configure "upstream" remote](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/configuring-a-remote-for-a-fork) and use `git rebase -i upstream/master` (after `git fetch -t upstream`)

> wants to merge 10 commits

Patch with single commit is preferable (multiple commits with merges may break GitHub UI for reviewing).

File: modules/gapi/include/opencv2/gapi/gkernel.hpp (hunk 1 above)
Comment (anna-plyakina) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp]: Applied.

File: modules/gapi/include/opencv2/gapi/gkernel.hpp [hunk 3]
```diff
@@ -233,29 +234,28 @@ class GKernelType<K, std::function<R(Args...)> >
...
     struct Class final: public cv::GKernelTypeM<Class, std::function API >, \
-                        public detail::G_ID_HELPER_CLASS(Class)
-// {body} is to be defined by user
+                        public G_ID_HELPER_CLASS(Class)
+    // {body} is to be defined by user
 
 #define G_API_OP   G_TYPED_KERNEL
 #define G_API_OP_M G_TYPED_KERNEL_M
 
+//! @endcond
```
Comment (dmatveev) [inline modules/gapi/include/opencv2/gapi/gkernel.hpp]: G_TYPED_KERNEL and other macros shouldn't be ingored in the Doxygen, it is part of public API

Comment (akhakim) [reply modules/gapi/include/opencv2/gapi/gkernel.hpp]: Applied.

**Pull request merged**